### ISSUE TITLE
Use pause hints to reduce cpu burn in blocking wait strategy

### ISSUE THREAD

**Issue by m0wfo** [tuplestream/swift-disruptor#2]
**Is your feature request related to a problem? Please describe.**
To reduce the number of redundant cpu cycles used in BlockingWaitStrategy's `waitFor()` function, it would be good to add a hint telling the caller progress may not be possible. The Java disruptor implementation uses the [onSpinWait](https://docs.oracle.com/javase/9/docs/api/java/lang/Thread.html#onSpinWait--) method which appeared with Java 9.

**Describe the solution you'd like**
Figure out a semantic equivalent of `onSpinWait` in Swift, or failing that figure out a nice way of introducing `__mm_sleep` or similar directly.

**Comment by hassila** [comment]
The easiest way might be to simply add a c library with relevant (x64/arm) inline assembly that calls the relevant cpu instruction (_mm_pause etc) - I don’t believe swift support inline asm yet.

**Comment by m0wfo** [comment]
@hassila yeah I couldn't find an obvious way of doing it either- guess the [c wrapper](https://github.com/tuplestream/swift-disruptor/blob/master/Sources/_Volatile/include/_Volatile.h) could be extended easily enough
